src/LegacyStrings.jl

```diff
@@ -97,4 +98,11 @@ using Compat
     else
         include("rep.jl")
     end
+
+    if isdefined(Base, :DirectIndexString)
```

Comment:
I think you'll have to define this earlier, since `DirectIndexString` is used above. You'll be able to check that once you'll have a version of Julia without it.

Reply:
moved